### PR TITLE
Support for non side effecting tree

### PR DESCRIPTION
**Pull request by ddrozdov**

### CODE REVIEW

File: gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala [hunk 1]
```diff
@@ -396,6 +396,8 @@ case class GremlinScala[End, Labels <: HList](traversal: GraphTraversal[_, End])
 
   def times(maxLoops: Int) = GremlinScala[End, Labels](traversal.times(maxLoops))
 
+  def tree() = GremlinScala[Tree[_], Labels](traversal.tree())
```
Comment (mpollmeier) [inline gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala]: doesn't make a huge difference, but would be good to pass on the type of tree:

`def tree[A]() = GremlinScala[Tree[A], Labels](traversal.tree())`

**Comment by mpollmeier** [comment]
can you provide a small example use case please, e.g. in `TraversalSpec`?

File: gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala (hunk 1 above)
Comment (ddrozdov) [inline gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala]: This won't work since `traversal.tree()` returns `GraphTraversal<S, Tree>`.

**Comment by ddrozdov** [comment]
Sure, see the update.
Unfortunately tinkerpop's `Tree` structure is a bit confusing. Probably you would have some time to implement a nice Scala wrapper for it ;)

**Comment by mpollmeier** [comment]
noted :)
thank you!

**Pull request merged**